### PR TITLE
Improve geometry handling in JS decoder

### PR DESCRIPTION
**Pull request by springmeyer**
This PR overhauls the geometry storage in the JS decoder in order to be able to correctly handle complex geometries like polygons with rings and multi-geometries.

Specifically:
 - GeoJSON support is finished in order to use GeoJSON as the testing format.
 - "@mapbox/point-geometry" and a `feature.loadGeometry` API is added in order to make the geometries compatible with Maplibre GL JS
 - The `js/src/mvtUtils.ts` was only being used inside tests, so the code was moved
 - Test coverage was improved for geometry decoding

### CODE REVIEW

File: js/src/data/Projection.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+export class Projection {
+    size: number;
+    x0: number;
+    y0: number;
+    s1: number;
+    s2: number;
+
+    constructor(x: number, y: number, z: number) {
+        const extent = 4096;
```
Comment (nyurik) [inline js/src/data/Projection.ts]: shouldn't this be per layer rather than hardcoded?

**Review by nyurik — APPROVED** [review]
one uncertain thing, but seems ok

File: js/src/data/Projection.ts (hunk 1 above)
Comment (springmeyer) [reply js/src/data/Projection.ts]: Nice catch! I was meaning to fix this, but had forgotten. Solved in 100644

Comment (springmeyer) [reply js/src/data/Projection.ts]: /cc @ebrelsford `feature.extent` is now available.

**Pull request merged**